### ISSUE TITLE
Implement ZIO.replicate

### ISSUE THREAD

**Issue by jdegoes** [zio/zio#1261]
A quite common idiom in ZIO is `List.fill(n)(effect)`, to replicate an effect `n` times.

However, this consumes unnecessary memory, since the effect is the same, there is no need to create `n` copies of it.

Instead, we can return an `Iterable` that _pretends_ to be that long, and whose iterator simply returns the same effect `n` times in a row.

It may be a challenge making this in a way that compiles on 2.12 and 2.13 simultaneously.

```scala
object ZIO {
  def replicate[R, E, A](n: Int)(effect: ZIO[R, E, A]): Iterable[ZIO[R, E, A]] = ???
}
```

Be sure to add delegations to this method inside the companion objects for the type aliases:

 - [ ] UIO
 - [ ] RIO
 - [ ] Task
 - [ ] IO

These methods will simply delegate to `ZIO.replicate`, with the appropriate type aliases used instead of `ZIO`.

**Comment by dorsev** [comment]
I'll take this

**Comment by jdegoes** [comment]
@dorsev 🙏

**Comment by ghostdogpr** [comment]
Closed by #1262
